cunumeric/module.py

```diff
@@ -5577,6 +5578,519 @@ def nancumsum(
     )
 
 
+@add_boilerplate("a")
+def nanargmax(
+    a: ndarray,
+    axis: Any = None,
+    out: Union[ndarray, None] = None,
+    *,
+    keepdims: bool = False,
+) -> ndarray:
+    """
+
+    Return the indices of the maximum values in the specified axis ignoring
+    NaNs. For empty arrays and all-NaN slices, a ValueError is raised. Warning:
+    the results cannot be trusted if a slice contains only NaNs and -Infs.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.
+    axis : int, optional
+        By default, the index corresponds to the flattened array, otherwise
+        along the specified axis.
+    out : ndarray, optional
+        If provided, the result will be inserted into this array. It should
+        be of the appropriate shape and dtype.
+    keepdims : bool, optional
+        If this is set to True, the axes which are reduced are left
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the array.
+
+    Returns
+    -------
+    index_array : ndarray[int]
+        Array of indices into the array. It has the same shape as `a.shape`
+        with the dimension along `axis` removed.
+
+    See Also
+    --------
+    numpy.nanargmin, numpy.nanargmax
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    if a.size == 0:
+        raise ValueError("attempt to get nanargmax of an empty sequence")
+
+    # Raising a ValueError here will block the runtime from proceeding further.
+    # Set check_value_error to False to disable the check
+    check_value_error = True
```

Comment:
Add a command-line flag here for this boolean flag so the user can change the behavior: https://github.com/nv-legate/cunumeric/blob/branch-23.07/cunumeric/settings.py#L28

Reply:
and I believe the default value should be `False`. Otherwise, every `nanargmax` or `nanargmin` call becomes a blocking operation.

Reply:
ok